src/lxc/lxccontainer.c

```diff
@@ -2270,8 +2270,10 @@ static bool add_to_array(char ***names, char *cname, int pos)
 
 	*names = newnames;
 	newnames[pos] = strdup(cname);
-	if (!newnames[pos])
+	if (!newnames[pos]) {
+		*names = (char**)realloc(*names, (pos) * sizeof(char *));
```

Comment:
This `realloc` can fail in which case `*names` must not be updated.
(I also don't think this is really necessary or would help all that much. For a further next successful `add_to_array()` call the `realloc` on top should end up being a nop anyway, and if the `strdup` fails it's easily possible the shrinking here fails, too.)

Reply:
Hi,
I took back this change. In the current implementation all places where `add_to_array()` fails break further execution and free an array instantly. However, if a usage of `add_to_array()` changes at some time and a new item is added after adding of another one has failed, it will cause a memory leak. See my explanation for shrinking in 'remove_from_array()'.